### PR TITLE
Gives engineers the RCD round start and nerfs its base abilities to compensate

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
- Gives engineers an RCD as part of their round start equipment
- RCD by default will build/deconstruct slower if you already have another one in progress. This can be upgraded with the new cooling upgrade disk. Reconstructing (anything that was there roundstart as per the destructive scan) doesn't have this downside, only construction/deconstruction.
- RCD construction effects can now be attacked in order to cancel them. This can be deterred with the anti-disruption upgrade disk.
- RCDs for nukies and whatever don't have these downsides
- The CE's roundstart RCD also doesn't have these downsides

## Why It's Good For The Game
Construction and reconstruction are currently one of the worst aspects of SS13--they are so slow and tedious that any sort of mass destruction goes unfixed for 10-30+ minutes, if fixed at all. This limits us because it means people don't want traitors to create large explosions, for instance--I do and so I think it's crucial that we fix construction.

Reconstruction has already been improved on the RCD with the destructive scans, but I see no reason to limit this to something so out of the way.

Ideally the RCD even gets more functionalities, like the ability to print stock parts (or having stock parts removed), etc, in order to lessen this burden. 

## Changelog
:cl:
add: Engineers now have an RCD round start.
balance: RCD construction/deconstruction effects can now be attacked in order to cancel them. You can get the anti-disruption upgrade disk to prevent this.
balance: RCD construction/deconstruction is now slower if you already have another effect up. This does not effect reconstruction.
balance: Both of the above effects do not effect the CE's roundstart RCD, nor any other RCDs such as combat RCDs.
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
@Fikou likes this

**Comment by iwishforducks** [comment]
im torn on whether or not i like this. reconstruction is ass and RCDs make it so much easier. but as you have correctly assessed RCDs are quite powerful.

i dont know if i like making it so you have to install more upgrade disks into the RCD- the already existing upgrade disks were obtuse.

the ability to attack constructing RCD things is really good. honestly don’t even think there should be an upgrade disk to remove that part of it. it’d be really hard to tell if someone *has* that upgrade or not until you try to click their constructs. the nerf to construction speed if you’re building multiple at a time is confusing. i don’t really understand the point of that. is it to stop roundstart mass wall/window griefing? if so, wouldn’t the other engineers have tools to clean up said griefing? or is it to solve something else?

**Comment by Fikou** [comment]
> the ability to attack constructing RCD things is really good. honestly don’t even think there should be an upgrade disk to remove that part of it. it’d be really hard to tell if someone _has_ that upgrade or not until you try to click their constructs

it makes their holograms red i ithnk

**Comment by iwishforducks** [comment]
> > the ability to attack constructing RCD things is really good. honestly don’t even think there should be an upgrade disk to remove that part of it. it’d be really hard to tell if someone _has_ that upgrade or not until you try to click their constructs
> 
> it makes their holograms red i ithnk

sounds good but also still kind of obtuse. i don’t know if many people would pick up on the difference. i personally think it should be consistent all around. overall not that big of a deal though, i think most would realize red = destructible. maybe.

**Comment by improvedname** [comment]
I thinj rcds lost alot of value and probaly will only be used to place walls/windows now since you dont no longer need to place rods down before you can place floortiles and now just floortile the hole uo

**Comment by SomeoneYouProbablyKnow** [comment]
What abount engi-Borg RCDs? How are they gonna function?

**Comment by Jackal-boop** [comment]
if goofball never finishes the RAT i guess this is a solution

**Comment by atlasle** [comment]
> if goofball never finishes the RAT i guess this is a solution

the rat is dumb and bloat they should just make that a mech thing

**Comment by Mothblocks** [comment]
> What abount engi-Borg RCDs? How are they gonna function?

I'm not sure, I can check later but since this is being test merged is there any chance you could check this before I do? :)

**Comment by Mothblocks** [comment]
> im torn on whether or not i like this. reconstruction is ass and RCDs make it so much easier. but as you have correctly assessed RCDs are quite powerful.
> 
> i dont know if i like making it so you have to install more upgrade disks into the RCD- the already existing upgrade disks were obtuse.
> 
> the ability to attack constructing RCD things is really good. honestly don’t even think there should be an upgrade disk to remove that part of it. it’d be really hard to tell if someone _has_ that upgrade or not until you try to click their constructs. the nerf to construction speed if you’re building multiple at a time is confusing. i don’t really understand the point of that. is it to stop roundstart mass wall/window griefing? if so, wouldn’t the other engineers have tools to clean up said griefing? or is it to solve something else?

- Uninterruptible effects are red (because the blue is going to be the most common and looks the best and red is the harshest counterpart)
- The point of the slowdown changes is to hurt the roundstart RCD as a combat tool. I'm happy about RCD being usable in combat but I think the roundstart one specifically should only suggest that it can be used as one, not be an effective one. This change in particular is meant to make it harder to do something like walling off security, with this change if you were to try to place walls in front of the doors the warden should have enough time to run out and interrupt it and arrest you or whatever. As a note on your last part of this comment, I'm not a fan of problems being done by X job being primarily solved by that same job.
- I'm completely fine overcorrecting on construction and adjusting back later

**Comment by Mothblocks** [comment]
> if goofball never finishes the RAT i guess this is a solution

I was against the RAT replacing the RCD to begin with so I would be happy with both tbh

**Comment by MrMelbert** [comment]
> Gross, this should definitely change. RCDs should be only a little bit above manual material placing if that, with reconstruction being cheaper. What's the current exchange rate?

In fairness it's only noticeable when constructing things which use a lot of materials, walls (and probably floors but I haven't checked) are unaffected. Top is the current rate.

**Comment by iwishforducks** [comment]
alright so I went through and got some common costs for RCDs.

first and foremost, there is a max of 160 materials in an RCD. combat RCDs have 500.

1 iron sheet gives 4mu
1 glass sheet gives 4mu
1 reinforced glass sheet gives 6mu
1 plasteel sheet gives 12mu
1 matter cartridge gives 40mu

... matter cartridges take 6 iron and 4 glass to print, which means they are 100% efficient. there is no point in printing them.

going to list the matter unit cost for the item, then list the manual construction cost, and then the RCD material efficiency.

grille - 4mu - 1 iron sheet - 100% efficiency
directional normal window - 6mu - 1 glass sheet - 66% efficiency 
directional reinforced window - 9mu - 1 reinforced glass sheet - 66% efficiency
fulltile normal window - 12mu - 2 glass sheets - 66% efficiency
fulltile reinforced window - 15mu - 2 reinforced glass sheets - 80% efficiency

wall - 16mu - 4 iron sheets - 100% efficiency

airlock - 16mu - 4 iron sheets - 100% efficiency
airlock (glass) - 20mu - 4 iron sheets 1 glass - 100% efficiency

machine - 25mu - 5 iron sheets - 80% efficiency
computer - 25mu - 5 iron sheets - 80% efficiency

reconstruction has a massive impact, cutting the cost of the item by a 1/8th. so if you make walls with an RCD on a reconstruction spot you are running at 800% efficiency.

for the most part, it seems like most items are efficient, barring windows for whatever reason. also building a wall on a wall girder cuts the cost of the girder. so it's still 100% efficient.

**Comment by Jackal-boop** [comment]
> > if goofball never finishes the RAT i guess this is a solution
> 
> the rat is dumb and bloat they should just make that a mech thing

i think you are silly

File: code/game/objects/items/rcd/RCD.dm [hunk 1]
```diff
@@ -621,7 +640,20 @@ GLOBAL_VAR_INIT(icon_holographic_window, init_holographic_window())
...
+	desc = "A higher-end model of the rapid construction device, prefitted with improved cooling and disruption prevention. Provided to the chief engineer."
+	upgrade = RCD_UPGRADE_ANTI_INTERRUPT | RCD_UPGRADE_NO_FREQUENT_USE_COOLDOWN
+	matter = 160
+	color = list(
+		0.3, 0.3, 0.7, 0.0,
+		1.0, 1.0, 0.2, 0.0,
+		-0.2, 0.0, 1.0, 0.0,
+		0.0, 0.0, 0.0, 1.0,
+		0.0, 0.0, 0.0, 0.0,
+	)
```
Comment (ZephyrTFA) [inline code/game/objects/items/rcd/RCD.dm:656]: Could turn this into a theft objective, since this will actually see use unlike the station blueprints

**Review by ZephyrTFA — APPROVED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
Bomb enjoyers 🤢

**Comment by Vincent983** [comment]
hey, it means you can bomb MORE per antagonist round without the shuttle being called (the shuttle will still be called the moment a bomb with 1 devastation is detonated)

File: code/game/objects/items/rcd/RCD.dm (hunk 1 above)
Comment (distributivgesetz) [reply code/game/objects/items/rcd/RCD.dm:656]: I hate how unused the blueprints are because they are actually super useful and I use them all the time

**Comment by TheVekter** [comment]
Anything that encourages people to actually fix issues instead of fucking off as soon as something blows up is alright in my book.

I feel like the RAT will still have a place after this, if it ever actually gets implemented.

**Comment by SomeoneYouProbablyKnow** [comment]
> > What abount engi-Borg RCDs? How are they gonna function?
> 
> I'm not sure, I can check later but since this is being test merged is there any chance you could check this before I do? :)

And then i did (Ping on Sybil still BAD). So anyway
* Borg RCD are new weaker RCDs but you can't upgrade them. Should probably make it so adding an upgrade disk (Or emag Or malf hacking) bring that RCD to the level of CE's.
* RCDs in vendors cost 1000 cr still and are a bit redundant (Useful for tiders and tators i guess). Consider putting a limited amount of high cost upgrade disks there for engis to grind towards instead of just waiting on RnD. And maybe some spare RCDs around engineering if it's shiftstart gear maybe
* The slowdown on building multiple things is so severe that it's easier to build them one at a time. I guess that's intended but very annoying actually using RCD for building things. Donno what to do about that, maybe nothing eh.

**Comment by Vincent983** [comment]
why not just allow borg upgrades to be printed in the exofab and allow you to put them in borgs like how you can with PKA upgrades?

> > > What abount engi-Borg RCDs? How are they gonna function?
> > 
> > 
> > I'm not sure, I can check later but since this is being test merged is there any chance you could check this before I do? :)
> 
> And then i did (Ping on Sybil still BAD). So anyway
> 
>     * Borg RCD use the new weaker RCD version but you can't upgrade them. Should probably make it so adding an upgrade disk to a borg (Or emag Or malf hacking) brings that RCD to the level of CE's.
>

**Comment by Vincent983** [comment]
Another thing, considering the fact that there are 6 RCD upgrades, but they're all unlocked by the same node, why not compact them into a single upgrade?

**Comment by Mothblocks** [comment]
> The slowdown on building multiple things is so severe that it's easier to build them one at a time. I guess that's intended but very annoying actually using RCD for building things. Donno what to do about that, maybe nothing eh.

FWIW the disk to remove the cooldown is significantly cheaper than the others. Do you think you could try buying it when you feel its necessary and see if it's still too much? (Not the cooldowns, but the time and effort spent to get the disk and use it, is it worth it?)

**Comment by Mothblocks** [comment]
@SomeoneYouProbablyKnow forgot to ping

**Pull request merged**